FirebaseInstallations/Source/Library/InstallationsIDController/FIRInstallationsIDController.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class FBLPromise<ValueType>;
+@class FIRInstallationsItem;
+
+/**
+ * The class is responsible for managing FID for a given `FIRApp`.
```

Comment:
you mean managing FIRInstallationsStore? Because I don't see any local cached FID info. Or should we have a local cached FIRInstallationsItem so we are not reconstruct the object every query.

Reply:
@chliangGoogle 
By managing I mean following:
- retrieving /saving locally stored FID
- registering a new FID using FIS API
- removing FID

The class is not fully implemented yet, it is just an intermediate part without API (sorry, forgot to add a TODO for that).

I'll add more description on the FID lifecycle to the class description later on, so it will be more clear.